### PR TITLE
Simplifies check whether the CI image should be rebuilt

### PR DESCRIPTION
**Pull request by potiuk**
Rather than counting changed layers in the image (which was
enigmatic, difficult and prone to some magic number) we rely now
on random file generated while building the image.

We are using the docker image caching mechanism here. The random
file will be regenerated only when the previous layer (which is
about installling Airflow dependencies for the first time) gets
rebuild. And for us this is the indication, that the building
the image will take quite some time. This layer should be
relatively static - even if setup.py changes the CI image is
designed in the way that the first time installation of Airflow
dependencies is not invalidated.

This should lead to faster and less frequent rebuild for people
using Breeze and static checks.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Together with just merged #12177 we should finally handle all the cases where Breeze correctly detects when it is faster to build image locally or whether to pull it first.

**Comment by potiuk** [comment]
I would love to get that in and work today on rock-solid image pulling. This is the last part of what I wanted to do with simplifying the detection of "pull vs. build" and with this in place I think I will finally be able to solve some of the remaining unnecessary rebuilts of Breeze. Looking forward to some reviewers :)

File: Dockerfile.ci [hunk 1]
```diff
@@ -132,7 +132,9 @@ ARG RUNTIME_APT_DEPS="\
       sqlite3 \
       tmux \
       unzip \
-      vim"
+      vim \
+      xxd"
+ENV RUNTIME_APT_DEP=${RUNTIME_APT_DEPS}
```
Comment (feluelle) [inline Dockerfile.ci:137]:
```suggestion
ENV RUNTIME_APT_DEPS=${RUNTIME_APT_DEPS}
```

File: scripts/ci/libraries/_build_images.sh [hunk 2]
```diff
@@ -427,14 +416,12 @@ function build_images::rebuild_ci_image_if_needed() {
...
-                build_images::get_local_image_info
-                if [[ ${LOCAL_MANIFEST_IMAGE_UNAVAILABLE:=} != "true" ]]; then
-                    build_images::compare_layers
-                else
-                    FORCE_PULL_IMAGES="true"
-                fi
+            build_images::get_remote_image_build_cache_indicator
+            if [[ ${REMOTE_DOCKER_REGISTRY_UNREACHABLE:=} != "true" && \
+                  ${LOCAL_MANIFEST_IMAGE_UNAVAILABLE:=} != "true" ]]; then
+                    build_images::compare_local_and_remote_build_cache_indicator
```
Comment (feluelle) [inline scripts/ci/libraries/_build_images.sh]:
```suggestion
                build_images::compare_local_and_remote_build_cache_indicator
```
Shouldn't it be indented less (i.e. 4 spaces from the beginning of the if statement)?

**Comment by potiuk** [comment]
Thanks @feluelle ! Seems I am getting worse with the typos :D

**Comment by potiuk** [comment]
Hey @feluelle. All Good it seems!

**Review by feluelle — APPROVED** [review]

**Comment by potiuk** [comment]
Hey @feluelle -> i did a few more modifications to make it even clearer to know what's going on and double-checked what happens for migration for those who have the current image/manifest. It should be rather smooth migration now :).

**Pull request merged**